app/routes/exercises.rb

```diff
@@ -66,6 +66,28 @@ class Exercises < Core
         redirect '/dashboard'
       end
 
+      post '/exercises/:key/request-for-help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may request help."
+          redirect '/'
+        end
```

Comment:
I'd be OK with not testing this, since the `exercise.request_help!` is tested (and is a no-op if you're not the correct user. The branch above is more about being friendly about the whole thing.